src/main/pages/developer-guide/databases.adoc

```diff
@@ -0,0 +1,177 @@
+    5.0.0-M1/
+        1__init.sql
+    5.0.0-M2
+        1__rename_tables.sql
+        2__insert_predefined_data.sql
+        postgresql/
+            2__insert_predefined_data.sql
+subproject-schema/
+    5.0.0-M1/
+        1.1__init_subproject.sql
+    5.0.0-M2
+        2.1__create_additional_tables.sql
+----
+
+Location directory
+The main root directory for scripts (in this example, `che-schema` and `subproject-schema` is used as location names for Flyway.
+There can be multiple locations (artifacts) with scripts, and Flyway can be configured with property to search for such locations in classpath
+
+db.schema.flyway.scripts.locations=classpath:che-schema,classpath:subproject-schema
+
+
+* Version directory*
+There are version directories under the `che-schema` like 5.0.0 or 5.0.0-M1 these directories contain
+scripts for specific versions
+SQL scripts will be placed under project version directories line 1.init.sql or 1.rename_fields.sql
+The naming of scripts is pretty simple: the first digit in the name is a script version in a project versions (it's `1` in `1__init.sql`)
+then description of changes (if necessary) init in 1__init.sql and then the file extension .sql in 1__init.sql
+
+*Vendor specific script*
+
+There is a directory in 5.0.0-M2 called `postresql` if current database provider is posgresql then
+the script from 5.0.0-M1/posgresql/2.add_workspace_constraint.sql will be used instead of 5.0.0-M1/2.add_workspace_constraint.sql, so basically if the same script name is provided in provider-specific directory then this script will be used instead
+
+So, the order of applying scripts be as following
+[width="100%",options="header",]
+|===
+|db version |script name	|location	|picked vendor specific
+|5.0.0.1.1 |1__init.sql	|che-schema	|no
+|5.0.0.1.1.1	|1.1__init_subproject.sql	|subproject-schema	|no
+|5.0.0.2.1	|1__rename_tables.sql	|che-schema	|no
+|5.0.0.2.2	|2__insert_predefined_data.sql	|che-schema	|yes
+|5.0.0.2.2.1	|2.1__create_new_tables.sql	|subproject-schema	|no
+|===
+
+[id="pg-trgm"]
+== pg_trgm
+Postgres Trigram extension is used for more optimised search of similar string https://www.postgresql.org/docs/9.6/static/pgtrgm.html
+
+in Che it is used for a faster search for similar email and names, and enabled with a vendor specific migration script:
+```
+CREATE EXTENSION IF NOT EXISTS pg_trgm;
+CREATE INDEX index_user_lower_email ON usr USING GIN (LOWER(email) gin_trgm_ops);
+CREATE INDEX index_user_lower_name ON usr USING GIN (LOWER(name) gin_trgm_ops);
+```
+
+[id="contributor-guidelines"]
+== Contributor guidelines
+*Creating a module to run TCK on a custom DB*
+
+In order to run TCK for a custom database, a maven module should be created with following components:
+
+- configure `docker-maven-plugin` to run an image with database
+- a Guice module that extends `TckModule` which will be responsible for establishing connection with DataSource, binding all required implementations of JPA entities and repositories.
```

Comment:
Some items are with dots at the end, some without. I suppose it's a common way to put `;` at the end of all items except the last, and the last item should end with `.`
WDYT?